js/hotgraphicPopupView.js

```diff
@@ -1,137 +1,137 @@
-            nextItem.toggleActive();
-            nextItem.toggleVisited(true);
-        }
-
-    });
-
-    return HotgraphicPopupView;
+  'use strict';
+
+  var HotgraphicPopupView = Backbone.View.extend({
+
+    className: 'hotgraphic-popup',
+
+    events: {
+      'click .js-hotgraphic-popup-close': 'closePopup',
+      'click .js-hotgraphic-controls-click': 'onControlClick'
+    },
+
+    initialize: function() {
+      this.listenToOnce(Adapt, "notify:opened", this.onOpened);
+      this.listenTo(this.model.get('_children'), {
+        'change:_isActive': this.onItemsActiveChange,
+        'change:_isVisited': this.onItemsVisitedChange
+      });
+      this.render();
+    },
+
+    onOpened: function() {
+      this.applyNavigationClasses(this.model.getActiveItem().get('_index'));
+      this.updatePageCount();
+      this.handleTabs();
+    },
+
+    applyNavigationClasses: function (index) {
+      var itemCount = this.model.get('_items').length;
+      var canCycleThroughPagination = this.model.get('_canCycleThroughPagination');
+
+      var shouldEnableBack = index > 0 || canCycleThroughPagination;
+      var shouldEnableNext = index < itemCount - 1 || canCycleThroughPagination;
+      var $controls = this.$('.hotgraphic-popup__controls');
+
+      this.$('hotgraphic-popup__nav')
+          .toggleClass('first', !shouldEnableBack)
+          .toggleClass('last', !shouldEnableNext);
+
+      $controls.filter('.back').a11y_cntrl_enabled(shouldEnableBack);
+      $controls.filter('.next').a11y_cntrl_enabled(shouldEnableNext);
+    },
+
+    updatePageCount: function() {
+      var template = Adapt.course.get("_globals")._components._hotgraphic.popupPagination || '{{itemNumber}} / {{totalItems}}';
+      var labelText = Handlebars.compile(template)({
+        itemNumber: this.model.getActiveItem().get('_index') + 1,
+        totalItems: this.model.get("_items").length
+      });
+      this.$('.hotgraphic-popup__count').html(labelText);
+    },
+
+    handleTabs: function() {
+      this.$('.hotgraphic-popup__item:not(.is-active) *').a11y_on(false);
+      this.$('.hotgraphic-popup__item.is-active *').a11y_on(true);
+    },
+
+    onItemsActiveChange: function(item, _isActive) {
+      if (!_isActive) return;
+      var index = item.get('_index');
+      this.updatePageCount();
+      this.handleTabs();
+      this.applyItemClasses(index);
+      this.handleFocus(index);
+    },
+
+    applyItemClasses: function(index) {
+      this.$('.hotgraphic-popup__item[data-index="' + index + '"]').addClass('is-active').removeAttr('aria-hidden');
+      this.$('.hotgraphic-popup__item[data-index="' + index + '"] .hotgraphic-popup__item-title').attr("id", "notify-heading");
+      this.$('.hotgraphic-popup__item:not([data-index="' + index + '"])').removeClass('is-active').attr('aria-hidden', 'true');
+      this.$('.hotgraphic-popup__item:not([data-index="' + index + '"]) .hotgraphic-popup__item-title').removeAttr("id");
```

Comment:
If you have time, please can we cache and chain in this function to make it legible:
```js
applyItemClasses: function(index) {
  var $items = this.$('.hotgraphic-popup__item');
  var $activeItem = $items.filter('[data-index="' + index + '"]');

  $activeItem.addClass('is-active').removeAttr('aria-hidden')
    .find('.hotgraphic-popup__item-title').attr('id', 'notify-heading');

  $items.not($activeItem).removeClass('is-active').attr('aria-hidden', 'true')
    .find('.hotgraphic-popup__item-title').removeAttr('id');
}
```